### PR TITLE
Remove old type-ignore explanation comments not removed in #8280

### PR DESCRIPTION
**Pull request by Avasam**
https://github.com/python/typeshed/issues/8280 removed some redundant `#type: ignore`, but not the associated comments.

### CODE REVIEW

File: stdlib/builtins.pyi [hunk 1]
```diff
@@ -98,7 +98,6 @@ class object:
     __annotations__: dict[str, Any]
     @property
     def __class__(self) -> type[Self]: ...
-    # Ignore errors about type mismatch between property getter and setter
```
Comment (Avasam) [inline stdlib/builtins.pyi:101]: Noticed by @Akuli in https://github.com/python/typeshed/pull/11496#discussion_r1508111612
Added in #5092

File: stdlib/weakref.pyi [hunk 2]
```diff
@@ -93,7 +93,6 @@ class WeakValueDictionary(MutableMapping[_KT, _VT]):
 
 class KeyedRef(ref[_T], Generic[_KT, _T]):
     key: _KT
-    # This __new__ method uses a non-standard name for the "cls" parameter
```
Comment (Avasam) [inline stdlib/weakref.pyi:96]: Added in #5092

File: stdlib/importlib/metadata/__init__.pyi [hunk 3]
```diff
@@ -241,7 +241,6 @@ class MetadataPathFinder(DistributionFinder):
     @classmethod
     def find_distributions(cls, context: DistributionFinder.Context = ...) -> Iterable[PathDistribution]: ...
     if sys.version_info >= (3, 10):
-        # Yes, this is an instance method that has a parameter named "cls"
```
Comment (Avasam) [inline stdlib/importlib/metadata/__init__.pyi:244]: Added in #6443

Comment (hauntsaninja) [reply stdlib/importlib/metadata/__init__.pyi:244]: Don't feel strongly, but this one is kind of funny and might help a human who is quickly glancing at stubs

**Review by hauntsaninja — APPROVED** [review]

File: stdlib/importlib/metadata/__init__.pyi (hunk 3 above)
Comment (Avasam) [reply stdlib/importlib/metadata/__init__.pyi:244]: I have no issue leaving it in if you feel it has value outside of explaining a `type: ignore` comment

**Pull request merged**